### PR TITLE
CB-1891 Improve thread handling in CB

### PR DESCRIPTION
**Pull request by lacikaaa**

### CODE REVIEW

File: flow/src/main/java/com/sequenceiq/flow/reactor/config/EventBusConfig.java [hunk 1]
```diff
@@ -48,13 +54,13 @@ private Dispatcher getEventBusDispatcher() {
...
-        MDCCleanerThreadPoolExecutor executorService = new MDCCleanerThreadPoolExecutor(eventBusThreadPoolSize,
-                eventBusThreadPoolSize,
+        MDCCleanerThreadPoolExecutor executorService = new MDCCleanerThreadPoolExecutor(eventBusThreadPoolCoreSize,
+                eventBusThreadPoolMaxSize,
                 0L,
                 TimeUnit.MILLISECONDS,
-                new LinkedBlockingQueue<>(eventBusThreadPoolSize),
+                new LinkedBlockingQueue<>(eventBusThreadPoolBacklogSize),
                 new NamedDaemonThreadFactory("reactorDispatcher", context),
                 (r, executor) -> r.run());
```
Comment (biharitomi) [inline flow/src/main/java/com/sequenceiq/flow/reactor/config/EventBusConfig.java]: I think this time we should define a proper `RejectedExecutionHandler` here to be able to track the lack of threads from the logs.

File: flow/src/main/java/com/sequenceiq/flow/reactor/config/EventBusConfig.java [hunk 2]
```diff
@@ -21,8 +21,14 @@
 @Configuration
 public class EventBusConfig {
 
-    @Value("${cb.eventbus.threadpool.core.size:50}")
-    private int eventBusThreadPoolSize;
+    @Value("${cb.eventbus.threadpool.core.size}")
+    private int eventBusThreadPoolCoreSize;
+
+    @Value("${cb.eventbus.threadpool.max.size:150}")
```
Comment (biharitomi) [inline flow/src/main/java/com/sequenceiq/flow/reactor/config/EventBusConfig.java]: I think we should add all of the eventbus's thread pool properties to the application.yml or none of them, but the mixed approach could be tricky if the pool size bigger then the max pool size :)

Comment (lacikaaa) [reply flow/src/main/java/com/sequenceiq/flow/reactor/config/EventBusConfig.java]: removed default value from application.yml

File: flow/src/main/java/com/sequenceiq/flow/reactor/config/EventBusConfig.java (hunk 1 above)
Comment (lacikaaa) [reply flow/src/main/java/com/sequenceiq/flow/reactor/config/EventBusConfig.java]: replaced with logging the error

**Pull request merged**